dockerfiles/Dockerfile

```diff
@@ -0,0 +1,170 @@
+# syntax=docker/dockerfile:1.3-labs
+ARG OPENMS_VERSION=3.1.0
+ARG OPENMS_LIBRARY_BUILD_DIR="/openms-build"
+ARG OPENMS_LIBRARY_DIR="/OpenMS"
+ARG CONTRIB_TAG=latest
+ARG CONTRIB_BASE_IMAGE=ghcr.io/openms/contrib:${CONTRIB_TAG}
+ARG CONTRIB_BUILD_DIR="/contrib-build"
+ARG THIRDPARTY_BRANCH="master"
+ARG THIRDPARTY_DIR="/thirdparty"
+ARG NUM_BUILD_CORES=8
+ARG MAKEFLAGS="-j${NUM_BUILD_CORES}"
+
+
+### OpenMS base ###
+FROM ${CONTRIB_BASE_IMAGE} as base
+ARG OPENMS_VERSION
+
+# the main reason we're pulling this in is to make sure that the metadata
+# below is consistently formatted for releases
+LABEL version="${OPENMS_VERSION}"
+LABEL software.version="${OPENMS_VERSION}-Ubuntu22.04"
+LABEL org.opencontainers.image.source https://github.com/OpenMS/OpenMS
+
+
+### OpenMS library ###
+FROM base as library
+ARG CONTRIB_BUILD_DIR
+ARG OPENMS_LIBRARY_DIR
+ARG OPENMS_LIBRARY_BUILD_DIR
+ARG MAKEFLAGS
+ENV MAKEFLAGS="${MAKEFLAGS}"
+
+# NOTE: this assumes that the context is root of the OpenMS repo
+COPY . ${OPENMS_LIBRARY_DIR}
+WORKDIR ${OPENMS_LIBRARY_BUILD_DIR}
+RUN /bin/bash -c "cmake -DCMAKE_BUILD_TYPE='Release' -DCMAKE_PREFIX_PATH='${CONTRIB_BUILD_DIR}/;/usr/;/usr/local' -DBOOST_USE_STATIC=OFF ${OPENMS_LIBRARY_DIR}"
+RUN make OpenMS
+LABEL software="OpenMS (library)"
+
+
+### OpenMS tools build ###
+FROM library as tools-build
+ARG OPENMS_LIBRARY_DIR
+ARG OPENMS_LIBRARY_BUILD_DIR
+ARG MAKEFLAGS
+ARG THIRDPARTY_BRANCH
+ARG THIRDPARTY_DIR
+ENV MAKEFLAGS="${MAKEFLAGS}"
+
+RUN apt-get update && apt-get install -y --no-install-recommends --no-install-suggests openjdk-17-jdk
+
+WORKDIR ${OPENMS_LIBRARY_DIR}
+RUN <<-EOF
+    mkdir /thirdparty
+    # note, we are not using the submodule here because the .git folder
+    # is not copied to this stage. Instead, we clone the repo directly.
+    git clone --branch=${THIRDPARTY_BRANCH} --depth=1 https://github.com/OpenMS/THIRDPARTY.git
+
+    # copying only the relevant binaries and leaving the rest behind
+    cp -r THIRDPARTY/All/* ${THIRDPARTY_DIR}
+    cp -r THIRDPARTY/Linux/64bit/* ${THIRDPARTY_DIR}
+
+    rm -rf THIRDPARTY
+EOF
+
+ENV PATH="${THIRDPARTY_DIR}/LuciPHOr2:${THIRDPARTY_DIR}/MSGFPlus:${THIRDPARTY_DIR}/Sirius:${THIRDPARTY_DIR}/ThermoRawFileParser:${THIRDPARTY_DIR}/Comet:${THIRDPARTY_DIR}/Fido:${THIRDPARTY_DIR}/MaRaCluster:${THIRDPARTY_DIR}/Percolator:${THIRDPARTY_DIR}/SpectraST:${THIRDPARTY_DIR}/XTandem:${THIRDPARTY_DIR}/Sage:${PATH}"
```

Comment:
Not sure but very recently we switched our support policy for Sirius. We only read their workspaces now and do not provide adapters anymore due to the size of their package. You might be able to remove it.

Reply:
Will you be removing it from the thirdparty repo?

Reply:
Yes I think so. Maybe we did already @axelwalter ?

Reply:
The changes regarding Sirius tools are in PR #7234. Once merged, I will remove Sirius from THIRDPARTY repository.